server/api/auth_handler.py

```diff
@@ -7,6 +7,24 @@
 auth_blueprint = Blueprint("auth_blueprint", __name__)
 
 
+# Wrapper that obtains and verifies an auth token in a cookie. Returns true if valid, false otherwise.
+def login_cookie_getter(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        try:
+            auth_header = request.cookies.get("Authentication token")
+            if auth_header:
+                auth_token = auth_header
+                auth_token = User.decode_auth_token(auth_token)
+                if not isinstance(auth_token, str):
+                    g.user = User.query.filter_by(id=auth_token).first()
+        except:
```

Comment:
In this case, let's handle this here in the decorator, rather than in the calling function. Return an error message to the client and do not call the wrapped function